src/ucs/sys/ptrcache.h

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2018.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_PTR_CACHE_H_
+#define UCS_PTR_CACHE_H_
+
+#include <ucs/datastruct/pgtable.h>
+#include <ucs/datastruct/list.h>
+#include <ucs/stats/stats_fwd.h>
+#include <ucm/api/ucm.h>
+
+typedef struct ucs_ptrcache         ucs_ptrcache_t;
+typedef struct ucs_ptrcache_region  ucs_ptrcache_region_t;
+
+
+struct ucs_ptrcache_region {
+    ucs_pgt_region_t    super;    /**< Base class - page table region */
+    ucs_list_link_t     list;     /**< List element */
+    ucm_mem_type_t      mem_type; /**< Memory type the pointer belongs to */
+};
+
+
+struct ucs_ptrcache {
```

Comment:
seems like the goal of this structure is to cache the memory type of non-host memory (per the events that UCM provides), so it should be called "ucs_memtype_cache"